lib/Server.js

```diff
@@ -677,25 +677,22 @@ class Server {
     const SocketServerImplementation = this.socketServerImplementation;
     this.socketServer = new SocketServerImplementation(this);
 
-    this.socketServer.onConnection((connection) => {
+    this.socketServer.onConnection((connection, headers) => {
       if (!connection) {
         return;
       }
 
-      if (
-        !this.checkHost(connection.headers) ||
-        !this.checkOrigin(connection.headers)
-      ) {
+      if (headers && (!this.checkHost(headers) || !this.checkOrigin(headers))) {
```

Comment:
This looks like it's too easy to accidentally skip the security check.

Maybe changing it to `if (!headers || ...`

Reply:
@sokra Good point. I think if users do not implement `onConnection` correctly though, it will be hard to figure out what is wrong. Maybe before that I'll log a warning if headers are missing.

Reply:
@Loonride it should be no warning, we should drop connection if headers are not present

Reply:
@evilebottnawi If `!headers` is true it means that the user made a socket server implementation that did not call the `onConnection(f)` callback correctly, so I think that could cause confusion if there is little explanation for failed connection. Let me send a PR and I'll show you what I mean

Reply:
:+1: